### PR TITLE
List amqp10 connections

### PR DESCRIPTION
**Pull request by MarcialRosales**
## Proposed Changes

This PR fixes issue https://github.com/rabbitmq/rabbitmq-server/issues/4238. 

The issue was due to how we searched for the reader process along the supervisor hierarchy tree. 
Additionally, we added the missing `internal_info` method that provided the info item `pid` which was missing from the reader module.

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

### CODE REVIEW

File: deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl [hunk 1]
```diff
@@ -187,6 +187,7 @@ handle_other({bump_credit, Msg}, State) ->
 handle_other(terminate_connection, State) ->
     State;
 handle_other({info, InfoItems, Pid}, State) ->
+    %logger:info("Reader info:  State: ~p pid ~p infoitems: ~p ", [State, Pid, InfoItems]),
```
Comment (kjnilsson) [inline deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl:190]: remove pls.

File: deps/rabbitmq_amqp1_0/test/command_SUITE.erl [hunk 2]
```diff
@@ -10,40 +10,50 @@
...
--include_lib("amqp_client/include/amqp_client.hrl").
 -include("rabbit_amqp1_0.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
 
 -define(COMMAND, 'Elixir.RabbitMQ.CLI.Ctl.Commands.ListAmqp10ConnectionsCommand').
 
 all() ->
     [
-     {group, non_parallel_tests}
+       {group, non_parallel_tests}
```
Comment (kjnilsson) [inline deps/rabbitmq_amqp1_0/test/command_SUITE.erl:20]: I am not sure about his indentation change. Doesn't matter too much though.

File: deps/rabbitmq_amqp1_0/test/command_SUITE.erl [hunk 3]
```diff
@@ -67,3 +77,89 @@ validate(_Config) ->
     ok = ?COMMAND:validate([atom_to_binary(K, utf8) || K <- ?INFO_ITEMS], #{}),
     {validation_failure,{bad_info_key,[other]}} =
         ?COMMAND:validate([<<"other">>], #{}).
+
+println(What, Value) -> io:format("~p : ~p~n", [What, Value]).
```
Comment (kjnilsson) [inline deps/rabbitmq_amqp1_0/test/command_SUITE.erl:81]: In test suites we normally prefer to use `ct:pal/2` for this kind of thing

File: deps/rabbitmq_amqp1_0/test/system_SUITE_data/console/Program.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+﻿using Amqp;
```
Comment (kjnilsson) [inline deps/rabbitmq_amqp1_0/test/system_SUITE_data/console/Program.cs:1]: Any particular reason not to just add this to the existing F# test program?

**Review by kjnilsson — CHANGES_REQUESTED** [review]

File: deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl (hunk 1 above)
Comment (MarcialRosales) [reply deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl:190]: absolutely. Forgot to remove it !

File: deps/rabbitmq_amqp1_0/test/system_SUITE_data/console/Program.cs (hunk 4 above)
Comment (MarcialRosales) [reply deps/rabbitmq_amqp1_0/test/system_SUITE_data/console/Program.cs:1]: Because I needed to keep an amqp 1.0 connection opened as long as I needed for exploratory testing. I found it very convenient to discover the supervisor hierarchy and to manually test rabbitmqctl command, etc. 
If I could open an amqp 1.0 connection via PerfTest I would have not written this little console app.

File: deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl [hunk 5]
```diff
@@ -763,11 +766,11 @@ info_internal(frame_max, #v1{connection = #v1_connection{frame_max = Val}}) ->
 info_internal(timeout, #v1{connection = #v1_connection{timeout_sec = Val}}) ->
     Val;
 info_internal(user,
-              #v1{connection = #v1_connection{user = #user{username = none}}}) ->
-    '';
-info_internal(username,
-              #v1{connection = #v1_connection{user = #user{username = Val}}}) ->
+              #v1{connection = #v1_connection{user = {user, Val, _, _}}}) ->
```
Comment (lukebakken) [inline deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl:769]: Why was this changed from using the `#user{}` record?

**Review by lukebakken — CHANGES_REQUESTED** [review]
@MarcialRosales thank you for this! I just have one question.

**Comment by lukebakken** [comment]
Testing results look good...

* Command output is as expected ✔️ 
    ```
    $ ./sbin/rabbitmqctl list_amqp10_connections
    Listing AMQP 1.0 connections ...
    pid
    <rabbit@shostakovich.1663893595.811.0>
    ```
* Viewing the connection from the management UI works as expected ✔️ 
* Closing the connection from the management UI works as expected ✔️

File: deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl (hunk 5 above)
Comment (MarcialRosales) [reply deps/rabbitmq_amqp1_0/src/rabbit_amqp1_0_reader.erl]: There were two changes. One was to remove `info_internal(username, ...)` given that `username` was not in the list of info items. And with regards the second change, which is what you are referring to, you are totally right. It is far more appropriate to use the original record expression. I have changed the code. Thanks @lukebakken

**Review by lukebakken — APPROVED** [review]

**Comment by ansd** [comment]
This PR introduces 2 new bugs:
1. It not only lists AMQP 1.0 connections, but also queries AMQP 0.9.1 connections.
2. Crashes occur when there are AMQP 0.9.1 connections.

To reproduce, run `make -C deps/rabbitmq_amqp1_0 ct-command t=non_parallel_tests:when_one_connection FULL=1` with https://github.com/rabbitmq/rabbitmq-server/pull/9290 enabled, or manually create 1 AMQP 0.9.1 connection and run `rabbitmqctl list_amqp10_connections` and check the RabbitMQ log file:
```
crasher:
  initial call: rabbit_reader:init/3
  pid: <0.652.0>
  registered_name: []
  exception exit: {unexpected_message,{info,[pid],<0.707.0>}}
    in function  rabbit_reader:handle_other/2 (rabbit_reader.erl, line 640)
    in call from rabbit_reader:mainloop/4 (rabbit_reader.erl, line 531)
    in call from rabbit_reader:run/1 (rabbit_reader.erl, line 453)
    in call from rabbit_reader:start_connection/5 (rabbit_reader.erl, line 352)
```
